views/components/main/parts/minishipitem.es

```diff
@@ -335,25 +166,28 @@ export class MiniShipRow extends Component {
           <div className="ship-item">
             { enableAvatar && (
               <Avatar mstId={$ship.api_id} isDamaged={hpPercentage <= 50} height={33}>
-                {compact ? <div className='ship-lv-avatar'>Lv. {level || '??'}</div> : null}
+                {compact ? <div className='ship-lv-avatar'>{levelString}</div> : null}
               </Avatar>
             ) }
             <OverlayTrigger placement='top' overlay={
               <Tooltip id={`miniship-exp-${ship.api_id}`}>
                 {
                   hideShipName ? (
                     <div className="ship-tooltip-info">
-                      <div>{$ship.api_name ? t(`resources:${$ship.api_name}`) : '??'}</div>
-                      {exp > 0 ? <div>Total: {exp}</div> : null}
-                      {nextExp > 0 ? <div>Next: {nextExp}</div> : null}
-                      {remodelExpString ? <div>{remodelExpString}</div> : null}
-                      {maxExpString ? <div>{maxExpString}</div> : null}
+                      <div>
+                        {$ship.api_name ? t(`resources:${$ship.api_name}`) : '??'}
+                      </div>
+                      <div>
+                        {levelString}
+                      </div>
+                      {exp > 0 ? <div>{t('main:Total')}: {exp}</div> : null}
+                      {nextExp > 0 ? <div>{t('main:Next')}: {nextExp}</div> : null}
+                      {remodelString ? <div>{remodelString}</div> : null}
                     </div>
                   ) : <div>
-                    {exp > 0 ? <div>Total: {exp}</div> : null}
-                    {nextExp > 0 ? <div>Next: {nextExp}</div> : null}
-                    {remodelExpString ? <div>{remodelExpString}</div> : null}
-                    {maxExpString ? <div>{maxExpString}</div> : null}
+                    {exp > 0 ? <div>{t('main:Total')}: {exp}</div> : null}
```

Comment:
React will render nothing if the content is `null` or `undefined` or `false`, you could use this to simplify your code:
```js
{exp > 0 && <div></div>}
```

Reply:
Ah, I see, there was
https://github.com/poooi/poi/blob/f006c177dceb22a45c2dcda2f522628dde5eaedf/views/components/main/parts/minishipitem.es#L163
so I used that.

Reply:
`remodelString` related code become a little complex, it could be simplified in that way

alongside with i18next, adding content could be easier